File: snow_pkg/src/openalea/SnowAccumulation.py
# coding: utf8
from copy import copy

import numpy
from math import *

def model_SnowAccumulation(tsmax = 0.0,
         tmax = 0.0,
         trmax = 0.0,
         precip = 0.0):
    """
     - Name: SnowAccumulation -Version: 1.0, -Time step: 1
     - Description:
                 * Title: snowfall accumulation  calculation
                 * Author: STICS
                 * Reference: -
                 * Institution: INRA
                 * Abstract: -
     - inputs:
                 * name: tsmax
                               ** parametercategory : constant
                               ** min : 0.0
                               ** datatype : DOUBLE
                               ** max : 1000
                               ** uri : 
                               ** default : 0.0
                               ** inputtype : parameter
                               ** unit : degC
                               ** description : maximum daily air temperature (tmax) below which all precipitation is assumed to be snow
                 * name: tmax
                               ** min : 0.0
                               ** default : 0.0
                               ** max : 5000.0
                               ** uri : 
                               ** variablecategory : auxiliary
                               ** datatype : DOUBLE
                               ** inputtype : variable
                               ** unit : degC
                               ** description : current maximum air temperature
                 * name: trmax
                               ** parametercategory : constant
                               ** min : 0.0
                               ** datatype : DOUBLE
                               ** max : 5000.0
                               ** uri : 
                               ** default : 0.0
                               ** inputtype : parameter
                               ** unit : degC
                               ** description : tmax above which all precipitation is assumed to be rain
                 * name: precip
                               ** min : 0.0
                               ** default : 0.0
                               ** max : 5000.0
                               ** uri : 
                               ** variablecategory : auxiliary
                               ** datatype : DOUBLE
                               ** inputtype : variable
                               ** unit : mmW
                               ** description : recalculated precipitation
     - outputs:
                 * name: Snowaccu
                               ** min : 0.0
                               ** variablecategory : rate
                               ** max : 500.0
                               ** uri : 
                               ** datatype : DOUBLE
                               ** unit : mmW/d
                               ** description : snowfall accumulation
    """

    fs = 0.0
    if tmax < tsmax:
        fs = 1.0
    if tmax >= tsmax and tmax <= trmax:
        fs = (trmax - tmax) / (trmax - tsmax)
    Snowaccu = fs * precip * 1
    return Snowaccu

File: snow_pkg/src/dssat/SnowAccumulation.f90
MODULE Snowaccumulationmod
    IMPLICIT NONE
CONTAINS

    SUBROUTINE model_SnowAccumulation(tsmax, &
        tmax, &
        trmax, &
        precip, &
        Snowaccu)
        IMPLICIT NONE
        REAL, INTENT(IN) :: tsmax
        REAL, INTENT(IN) :: tmax
        REAL, INTENT(IN) :: trmax
        REAL, INTENT(IN) :: precip
        REAL, INTENT(OUT) :: Snowaccu
        REAL:: fs
        fs = 0.0
        !- Description:
    !            * Title: snowfall accumulation  calculation
    !            * Author: STICS
    !            * Reference: -
    !            * Institution: INRA
    !            * Abstract: -
        !- inputs:
    !            * name: tsmax
    !                          ** description : maximum daily air temperature (tmax) below which all precipitation is assumed to be snow
    !                          ** inputtype : parameter
    !                          ** parametercategory : constant
    !                          ** datatype : DOUBLE
    !                          ** default : 0.0
    !                          ** min : 0.0
    !                          ** max : 1000
    !                          ** unit : degC
    !                          ** uri : 
    !            * name: tmax
    !                          ** description : current maximum air temperature
    !                          ** inputtype : variable
    !                          ** variablecategory : auxiliary
    !                          ** datatype : DOUBLE
    !                          ** default : 0.0
    !                          ** min : 0.0
    !                          ** max : 5000.0
    !                          ** unit : degC
    !                          ** uri : 
    !            * name: trmax
    !                          ** description : tmax above which all precipitation is assumed to be rain
    !                          ** inputtype : parameter
    !                          ** parametercategory : constant
    !                          ** datatype : DOUBLE
    !                          ** default : 0.0
    !                          ** min : 0.0
    !                          ** max : 5000.0
    !                          ** unit : degC
    !                          ** uri : 
    !            * name: precip
    !                          ** description : recalculated precipitation
    !                          ** inputtype : variable
    !                          ** variablecategory : auxiliary
    !                          ** datatype : DOUBLE
    !                          ** default : 0.0
    !                          ** min : 0.0
    !                          ** max : 5000.0
    !                          ** unit : mmW
    !                          ** uri : 
        !- outputs:
    !            * name: Snowaccu
    !                          ** description : snowfall accumulation
    !                          ** variablecategory : rate
    !                          ** datatype : DOUBLE
    !                          ** min : 0.0
    !                          ** max : 500.0
    !                          ** unit : mmW/d
    !                          ** uri : 
        IF(tmax .LT. tsmax) THEN
            fs = 1.0
        END IF
        IF(tmax .GE. tsmax .AND. tmax .LE. trmax) THEN
            fs = (trmax - tmax) / (trmax - tsmax)
        END IF
        Snowaccu = fs * precip * 1
    END SUBROUTINE model_SnowAccumulation

END MODULE
